### ISSUE TITLE
Can't add more on dependent dropdown

### ISSUE THREAD

**Issue by deshario** [unclead/yii2-multiple-input#261]
I m gonna implement Kartik DepDrop with this lib ... The result is the dependent is working but it occurs some error like image below ... when i opened page at first time but **dependent is working n i cant click plus button** Is there any solution to use this two lib perfectly

![capture](https://user-images.githubusercontent.com/11426727/49712470-2e20a800-fc77-11e8-90ed-966c8e563511.PNG)

**Form**
> 
    <?= $form->field($model, 'field')->widget(MultipleInput::className(), [
    'columns' => [
        [
            'name' => 'category',
            'type' => 'dropDownList',
            'title' => 'Category',
            'items' => $model->ha(),
            'options' => [
                'id' => 'item1',
                'prompt' => 'กรุณาเลือกยุทธศาสตร์',
            ],
        ],
        [
            'name' => 'chilly',
            'title' => 'paoprasong',
            'enableError' => true,
            'type' => \kartik\depdrop\DepDrop::class,
            'options' => [
                'id' => 'item2',
                'pluginOptions' => [
                    'depends' => ['item1'], // !!!! NOT replaced
                    'placeholder' => 'เลือกเปาประสงค์...',
                    'url' => Url::to(['/project/getgoal']),
                ],
            ],
        ],
    ],]); ?>

**Comment by unclead** [comment]
I haven't used DepDrop, could you explain what the problem is and what you want to achieve?

May be you have better use a closure to generate column options depending on current row data

**Comment by InsaneSkull** [comment]
https://github.com/unclead/yii2-multiple-input/wiki/Tips-and-Tricks#use-of-a-widgets-placeholder

**Comment by deshario** [comment]
@unclead the problem is when i load this page its error like in that image ... but the Depdrop works ... and the problem is i cant click plus button (add more multiple input)

**Comment by deshario** [comment]
@InsaneSkull  I had already tried that ... its not working ..

**Comment by unclead** [comment]
@deshario did you figure out why it does not work? we will be able to find a solution faster, knowing a reason

To be honest if you caught an error and you want to be this issue fixed asap you have better to debug it deeper by yourself to give us more details

**Comment by mabentley85** [comment]
@unclead I am experiencing the same issues when trying to use the [DepDrop plugin](https://github.com/kartik-v/dependent-dropdown).

In my use case, I have a parent dropdown and a dependent dropdown in each row.  Part of the configuration for the dependent dropdown is that it needs to be pointed to the parent dropdown's ID (http://plugins.krajee.com/dependent-dropdown#option-depends).  Currently nothing is working for me to set the dependent dropdown's parent ID as new rows are created during page render via PHP or in the browser using JavaScript.

I hope this clarifies things.

**Comment by unclead** [comment]
@mabentley85  it was fixed in 3d4895cb35d45535d554bccd781e8fddfc87820b
you can update to latest version (dev-master) to check that the issues is gone

i tried the following configuration
```
  [
            'name'  => 'user_id',
            'type'  => MultipleInputColumn::TYPE_DROPDOWN,
            'items' =>  [
                31 => 'item 31',
                32 => 'item 32',
                33 => 'item 33',
                34 => 'item 34',
                35 => 'item 35',
                36 => 'item 36',
            ],
            'options' => [
                'id' => 'user-id-{multiple_index_example1}'
            ]
        ],
        [
            'name' => 'owner_id',
            'type' => \kartik\depdrop\DepDrop::class,
            'options' => [
                'pluginOptions' => [
                    'depends' => ['user-id-{multiple_index_example1}'],
                    'url' => \yii\helpers\Url::to(['/site/dep-drop']),
                ],
            ],
        ],
```
as you can see i followed the instruction described here: https://github.com/unclead/yii2-multiple-input/wiki/Tips-and-Tricks#use-of-a-widgets-placeholder

**Comment by papppeter** [comment]
Hi,

this part won't work if we use a widget like select2. could this be fixed maybe?

`protected function renderWidget($type, $name, $value, $options)
    {
        unset($options['tabindex']);

        $model = $this->getModel();
        if ($model instanceof Model) {
            $widgetOptions = [
                'model'     => $model,
                'attribute' => $this->name,
                'value'     => $value,
                'options'   => [
                    'id'        => $this->normalize($name),
                    'name'      => $name,
                    'tabindex'  => self::TABINDEX,
                    'value'     => $value
                ]
            ];
        } else {
            $widgetOptions = [
                'name'      => $name,
                'value'     => $value,
                'options'   => [
                    'id'        => $this->normalize($name),
                    'name'      => $name,
                    'tabindex'  => self::TABINDEX,
                    'value'     => $value
                ]
            ];
        }

        $options = ArrayHelper::merge($options, $widgetOptions);

        return $type::widget($options);
    }`

$widgetOptions will overwrite the id we set for the widget.
